### ISSUE TITLE
[0.19] Failures on non-x86 architectures

### ISSUE THREAD

**Issue by olebole** [gammapy/gammapy#3662]
**Gammapy version**

0.19

**Bug description**

The recent version of gammapy fails on all Debian 64-bit platforms except x86_64. 

**Expected behavior**

All tests should pass.

**To Reproduce**

Running the CI test on a non-x86 platform, f.e. on arm64 gives:

```
______________________________ test_binary_erode _______________________________

    def test_binary_erode():
        geom = WcsGeom.create(binsz=0.02, width=2 * u.deg)
        mask = geom.region_mask("icrs;circle(0, 0, 1)")
    
        mask = mask.binary_erode(width=0.2 * u.deg, kernel="disk", use_fft=False)
        assert_allclose(mask.data.sum(), 4832)
    
        mask = mask.binary_erode(width=0.2 * u.deg, kernel="box", use_fft=True)
>       assert_allclose(mask.data.sum(), 3372)
E       AssertionError: 
E       Not equal to tolerance rtol=1e-07, atol=0
E       
E       Mismatched elements: 1 / 1 (100%)
E       Max absolute difference: 15
E       Max relative difference: 0.0044484
E        x: array(3387)
E        y: array(3372)

gammapy/maps/wcs/tests/test_ndmap.py:836: AssertionError
______________________________ test_binary_dilate ______________________________

    def test_binary_dilate():
        geom = WcsGeom.create(binsz=0.02, width=2 * u.deg)
        mask = geom.region_mask("icrs;circle(0, 0, 0.8)")
    
        mask = mask.binary_dilate(width=0.2 * u.deg, kernel="disk", use_fft=False)
        assert_allclose(mask.data.sum(), 8048)
    
        mask = mask.binary_dilate(width=(10, 10), kernel="box")
>       assert_allclose(mask.data.sum(), 9203)
E       AssertionError: 
E       Not equal to tolerance rtol=1e-07, atol=0
E       
E       Mismatched elements: 1 / 1 (100%)
E       Max absolute difference: 11
E       Max relative difference: 0.00119526
E        x: array(9214)
E        y: array(9203)

gammapy/maps/wcs/tests/test_ndmap.py:847: AssertionError
```

**Other information**

This has been observed so far on arm64, ppc64el, s390x, ppc64. It does not happen on 32-bit platforms (but they show other failures, update on https://github.com/gammapy/gammapy/issues/1993#issuecomment-979240280).

**Comment by adonath** [comment]
Thanks @olebole, I think this points to a real issue. In this specific case the methods use an FFT based algorithm, which might be affected by noise. In general the idea of treating binary operations with FFT was probably not great, however the solution is simple, we can just rely on https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.binary_erosion.html and https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.binary_dilation.html#scipy.ndimage.binary_dilation instead.

**Comment by adonath** [comment]
Solved by #3907.
